### PR TITLE
Comments clean up for inheritance margin

### PR DESCRIPTION
**Pull request by Cosifne**
Commits name indicates all the code change

### CODE REVIEW

File: src/EditorFeatures/CSharp/InheritanceMargin/CSharpInheritanceMarginService.cs [hunk 1]
```diff
@@ -42,7 +42,8 @@ protected override ImmutableArray<SyntaxNode> GetMembers(IEnumerable<SyntaxNode>
                     if (member.IsKind(
                         SyntaxKind.MethodDeclaration,
                         SyntaxKind.PropertyDeclaration,
-                        SyntaxKind.EventDeclaration))
```
Comment (Cosifne) [inline src/EditorFeatures/CSharp/InheritanceMargin/CSharpInheritanceMarginService.cs:45]: Tracking comment: https://github.com/dotnet/roslyn/pull/52357#discussion_r606352767

File: src/VisualStudio/Core/Def/ServicesVSResources.resx [hunk 2]
```diff
@@ -1710,4 +1710,20 @@ I agree to all of the foregoing:</value>
...
+  </data>
+  <data name="Navigate_to_0" xml:space="preserve">
+    <value>Navigate to '{0}'</value>
+  </data>
+  <data name="_0_in_1" xml:space="preserve">
+    <value>'{0}' in '{1}'</value>
+  </data>
+  <data name="Multiple_members_are_inherited_on_line_0" xml:space="preserve">
+    <value>Multiple members are inherited on line {0}</value>
+    <comment>Line number info is needed for accessibility purpose.</comment>
```
Comment (Cosifne) [inline src/VisualStudio/Core/Def/ServicesVSResources.resx:1727]: Orginal comment: https://github.com/dotnet/roslyn/pull/52357#discussion_r606353829

File: src/VisualStudio/Core/Def/Implementation/InheritanceMargin/InheritanceGlyphFactoryProvider.cs [hunk 3]
```diff
@@ -41,13 +42,18 @@ internal class InheritanceGlyphFactoryProvider : IGlyphFactoryProvider
...
         }
 
         public IGlyphFactory GetGlyphFactory(IWpfTextView view, IWpfTextViewMargin margin)
         {
-            return new InheritanceGlyphFactory(_threadingContext, _streamingFindUsagesPresenter, _classificationTypeMap, _classificationFormatMap, _waitIndicator);
+            return new InheritanceGlyphFactory(
+                _threadingContext,
+                _streamingFindUsagesPresenter,
+                _classificationTypeMap,
+                _classificationFormatMapService.GetClassificationFormatMap("tooltip"),
```
Comment (Cosifne) [inline src/VisualStudio/Core/Def/Implementation/InheritanceMargin/InheritanceGlyphFactoryProvider.cs:55]: Original comment: https://github.com/dotnet/roslyn/pull/52357#discussion_r606355028

File: src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginViewModel.cs [hunk 4]
```diff
@@ -60,13 +62,20 @@ internal class InheritanceMarginViewModel
...
-                var textAppended = " " + EditorFeaturesWpfResources.is_inherited;
+                // var textAppended = " " + ServicesVSResources.is_inherited;
+
+                // Here we want to show a classified text with loc text,
+                // e.g. 'Bar' is inherited.
+                // But the classified text are inlines, so can't directly use string.format to generate the string
                 var inlines = member.DisplayTexts.ToInlines(classificationFormatMap, classificationTypeMap);
-                WrapMemberWithinApostrophe(inlines);
-                inlines.Add(new Run(textAppended));
+                var startOfThePlaceholder = ServicesVSResources._0_is_inherited.IndexOf("{0}", StringComparison.Ordinal);
```
Comment (Cosifne) [inline src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginViewModel.cs:71]: Original comment: https://github.com/dotnet/roslyn/pull/52357#discussion_r606353367

File: src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMargin.xaml.cs [hunk 5]
```diff
@@ -82,8 +75,8 @@ private void TargetMenuItem_OnClick(object sender, RoutedEventArgs e)
 
         private void ChangeBorderToHoveringColor()
         {
-            SetResourceReference(BackgroundProperty, "VsBrush.CommandBarMenuBackgroundGradient");
-            SetResourceReference(BorderBrushProperty, "VsBrush.CommandBarMenuBorder");
+            SetResourceReference(BackgroundProperty, VsBrushes.CommandBarMenuBackgroundGradientKey);
```
Comment (Cosifne) [inline src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMargin.xaml.cs:78]: Orignal comment https://github.com/dotnet/roslyn/pull/52357#discussion_r606357467

File: src/EditorFeatures/Core.Wpf/Microsoft.CodeAnalysis.EditorFeatures.Wpf.csproj [hunk 6]
```diff
@@ -37,7 +37,6 @@
     <PackageReference Include="Microsoft.VisualStudio.Text.UI.Wpf" Version="$(MicrosoftVisualStudioTextUIWpfVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.SDK.EmbedInteropTypes" Version="$(MicrosoftVisualStudioSDKEmbedInteropTypesVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Threading" Version="$(MicrosoftVisualStudioThreadingVersion)" />
-    <PackageReference Include="Microsoft.VisualStudio.Shell.15.0" Version="$(MicrosoftVisualStudioShell150Version)" />
```
Comment (Cosifne) [inline src/EditorFeatures/Core.Wpf/Microsoft.CodeAnalysis.EditorFeatures.Wpf.csproj:40]: Original comment:
https://github.com/dotnet/roslyn/pull/52357#discussion_r606358199
I have move the whole UI part to Microsoft.VisualStudio.LanguageServices project

File: src/EditorFeatures/Core/FindUsages/IDefinitionsAndReferencesFactory.cs [hunk 7]
```diff
@@ -77,7 +77,7 @@ internal static class DefinitionItemExtensions
         {
             return ToDefinitionItemAsync(
                 definition, solution, isPrimary: false, includeHiddenLocations, includeClassifiedSpans: false,
-                options: FindReferencesSearchOptions.Default, cancellationToken: cancellationToken);
+                options: FindReferencesSearchOptions.Default.With(unidirectionalHierarchyCascade: true), cancellationToken: cancellationToken);
```
Comment (Cosifne) [inline src/EditorFeatures/Core/FindUsages/IDefinitionsAndReferencesFactory.cs:80]: Original comment: https://github.com/dotnet/roslyn/pull/52357#discussion_r606359746

File: src/EditorFeatures/CSharp/InheritanceMargin/CSharpInheritanceMarginService.cs [hunk 8]
```diff
@@ -67,6 +68,7 @@ protected override SyntaxToken GetIdentifier(SyntaxNode declarationNode)
                 EventDeclarationSyntax eventDeclarationNode => eventDeclarationNode.Identifier,
                 VariableDeclaratorSyntax variableDeclaratorNode => variableDeclaratorNode.Identifier,
                 TypeDeclarationSyntax baseTypeDeclarationNode => baseTypeDeclarationNode.Identifier,
+                IndexerDeclarationSyntax indexerDeclarationNode => indexerDeclarationNode.ThisKeyword,
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharp/InheritanceMargin/CSharpInheritanceMarginService.cs:71]: This is not an identifier :)

**Review by sharwell — APPROVED** [review]

File: src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginViewModel.cs [hunk 9]
```diff
@@ -60,13 +62,20 @@ internal class InheritanceMarginViewModel
...
+                var prefixString = ServicesVSResources._0_is_inherited[..startOfThePlaceholder];
+                var suffixString = ServicesVSResources._0_is_inherited[(startOfThePlaceholder + "{0}".Length)..];
+                inlines.Insert(0, new Run(prefixString));
+                inlines.Add(new Run(suffixString));
                 var toolTipTextBlock = inlines.ToTextBlock(classificationFormatMap);
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginViewModel.cs:76]: you may need to isnert a ltr marker to ensure that our code shows up property left-to-right, even when inserted into an rtl language.

File: src/EditorFeatures/CSharp/InheritanceMargin/CSharpInheritanceMarginService.cs (hunk 8 above)
Comment (Cosifne) [reply src/EditorFeatures/CSharp/InheritanceMargin/CSharpInheritanceMarginService.cs:71]: : ) Let me think, ok I would call it GetDeclarationToken()

File: src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginViewModel.cs (hunk 9 above)
Comment (Cosifne) [reply src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginViewModel.cs:77]: Good to know new things in WPF : )

**Pull request merged**